gatsby-node.js

```diff
@@ -8,229 +8,178 @@
-  // Allow requires from the src/ folder
-  config.merge({
-    resolve: {
-      root: path.join(__dirname, 'src')
-    },
-    target: 'web'
-  })
-
-  // Fix Gatsby setting `resolve.modulesDirectories` to `path.resolve(__dirname, "node_modules")`
-  // which causes module resolution errors when the npm tree is deduped
-  // See https://github.com/webpack/webpack/issues/6538#issuecomment-367324775
-  config.merge((current) => {
-    current.resolve.modulesDirectories = ['node_modules']
-
-    return current
-  })
-
-  // Allow requires from the src/ folder for postcss
-  config.merge((current) => {
-    const importPath = [
-      path.join(__dirname, 'src')
-    ]
-
-    if (typeof current.postcss === 'function') {
-      current.postcss = wrap(current.postcss, (postcss, wp) => [
-        require('postcss-import')({ addDependencyTo: wp, path: importPath }),
-        ...postcss(wp).slice(1, 3),
-        // Fix some weird reporting errors, see https://github.com/gatsbyjs/gatsby/issues/673#issuecomment-344245406
-        // This makes sure that the most recent updated `postcss-reporter` is used
-        require('postcss-reporter')
-      ])
-    } else if (current.postcss) {
-      current.postcss[0] = require('postcss-import')({ path: importPath })
-    }
+const SvgStorePlugin = require('external-svg-sprite-loader')
+const { defaultLocale, availableLocales } = require('./intl/config')
 
-    return current
-  })
+module.exports.onCreateWebpackConfig = ({ actions, getConfig, stage }) => {
+  let config = getConfig()
 
-  // Setup Babel & PostCSS
-  config.merge((current) => {
-    config.loader('js', (current) => {
-      current.query = {
-        ...current.query,
-        presets: [
-          [require.resolve('babel-preset-moxy'), {
-            react: true,
-            modules: 'commonjs',
-            targets: {
-              browsers: program.browserslist
-            }
-          }]
-        ],
-        plugins: [
-          require.resolve('react-hot-loader/babel'),
-          require.resolve('gatsby/dist/utils/babel-plugin-extract-graphql')
-        ]
-      }
+  // The default gatsby config perfer resolving dependencies to the root `node_modules`
+  // This is wrong and was causing problems when the same dependency but with different versions were installed
+  // Webpack should prefer resolving a sub-dependency starting from its deep node_modules folder and go up in the tree
+  // See https://github.com/gatsbyjs/gatsby/blob/a0cbbcb5519e53d2358a06be2c1a6ca0688280b7/packages/gatsby/src/utils/webpack.config.js#L363
+  // Furthermore, we add the `src` folder so that we can import internal modules without having to use relative paths
+  config.resolve.modules = ['node_modules', path.join(__dirname, 'src')]
 
-      return current
-    })
+  config.resolve.alias = {
+    ...config.resolve.alias,
+    // Add an alias to 'intl' so that we can import the intl config from within the src/
+    intl: path.join(__dirname, 'intl')
```

Comment:
Can we get rid of this and use `shared/intl-config` like before? I think it was more obvious. It was probably me who said to change this but nvm me :D